### PR TITLE
Audit listing with format json returns json, not a string

### LINKED ISSUE #6775: vault audit list -format json does not return JSON when list is empty
**Describe the bug**
The command ```vault audit list -format json``` does not consistently return a valid JSON object; if there are no audits enabled, the returned result is simply a string. When trying to automate certain steps, always receiving a JSON object is mandatory.

**To Reproduce**
1. make sure no audits are enabled (```vault audit disable ***```)
2. run ```vault audit list -format json```
3. you will see the following: 
```No audit devices are enabled.```

**Expected behavior**
I would expect to see an empty JSON object: ```{}```

**Environment:**
* Vault Server Version (retrieve with `vault status`):
```
# vault status
Key             Value
---             -----
Seal Type       shamir
Initialized     true
Sealed          false
Total Shares    5
Threshold       3
Version         1.1.2
Cluster Name    vault-cluster-3e8e9591
Cluster ID      7110de04-06b4-e73d-c2ce-94ac9616cde6
HA Enabled      false
```

* Vault CLI Version (retrieve with `vault version`):
```
Vault v1.1.2 ('0082501623c0b704b87b1fbc84c2d725994bac54')
```

* Server Operating System/Architecture:
```
# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.2 LTS
Release:	18.04
Codename:	bionic
```

* Vault server configuration file(s):
```
# cat /etc/vault.d/vault.hcl
api_addr = "http://127.0.0.1:8201"
ui = true

backend "file" {
    path = "/opt/vault"
}

listener "tcp" {
    address = "127.0.0.1:8200"
    tls_disable = 1
}
```

**Additional context**
None

### PR DESCRIPTION
**Pull request by jefferai**
Fixes #6775

### CODE REVIEW

**Comment by jefferai** [comment]
Sure, feel free to edit the branch and push more up to this PR.

File: command/audit_list.go [hunk 1]
```diff
@@ -94,13 +94,13 @@ func (c *AuditListCommand) Run(args []string) int {
...
-	if len(audits) == 0 {
-		c.UI.Output(fmt.Sprintf("No audit devices are enabled."))
-		return 0
-	}
-
 	switch Format(c.UI) {
 	case "table":
+		if len(audits) == 0 {
+			c.UI.Output(fmt.Sprintf("No audit devices are enabled."))
+			return 0
```
Comment (michelvocks) [inline command/audit_list.go]: `audit list` returns `0` as error code but literally any other list command returns `2`. Should we align that?

Comment (jefferai) [reply command/audit_list.go]: Huh. What does normal vault list do for empty?

Comment (michelvocks) [reply command/audit_list.go]: https://github.com/hashicorp/vault/blob/master/command/list.go#L87
https://github.com/hashicorp/vault/blob/master/command/namespace_list.go#L76

All other list commands return the error code `2` only the audit list command returns `0`. I'm thinking about aligning the return codes and also return the error code `2` here?

Comment (jefferai) [reply command/audit_list.go]: Sure, return 2

**Pull request merged**